### DISCUSSION TITLE
Cannot add models using Open WebUI

### DISCUSSION THREAD

**Discussion by factor3** [open-webui/open-webui#5426] [General]
Discussion https://github.com/open-webui/open-webui/discussions/7054

Go to 

```
Admin Panel
    ➡️ Settings
        ➡️Connections
             ➡️ Manage Ollama API connections
                 ➡️ Manage (🔧 icon) 
```
![image](https://github.com/user-attachments/assets/1aed8eb8-bc90-4b35-af70-fc8e22711152)

**Comment by peuportier** [comment]
Hello @factor3  You might first go terminal or command prompt . then run Ollama pull llama3.1:70b or direct from model place Admin - Model - Check if you can pull model here.

**Comment by factor3** [comment]
I believe I solved the problem.

It was necessary to use "ollama pull" to add a model into ollama, and once it was there it was added to the "base model" list. Once I did that, I was able to get the model(s) that I needed.

Thanks to peuportier for your suggestion.

**Reply by peuportier** [reply]
I'm happy that it's help. wish you a good continuation with OWUI

**Comment by MirkoBoeddecker** [comment]
I have the same problem. Openwebui is installed but I have no base modell. I pulled one from the command line but that did not change anything. How do I insert my first base modell and why is this made mandatory without ANY explanation?

**Reply by peuportier** [reply]
Can you be more specific my friend. What OS are you using? How did you install OWUI. Docker or not? You need to bring as much information you can for the community at least can understand your problem and help you the best we can

**Reply by MirkoBoeddecker** [reply]
I followed this: https://docs.openwebui.com/getting-started/#quick-start-with-docker- and needed some additional help from ChatGpt. I managed to fully install the gui. Everything is up and running in Docker on a Linux server. My problem is that if you do not have a base model you cannot do anything and there are no explanations how to install a base model. Docker and the gui are running on Ubuntu server 22.xx and I am limited to the command line on the server. I am accesing from a mac through the browser. 
Evereything runs as it seemingly should just that i can´t get started without having a base model and it seem I cannot set this up through the gui which is a downside and there are no expalantions whatsoever how to do this on the command line.

**Reply by peuportier** [reply]
@MirkoBoeddecker Here you can do for Install well on Ubuntu . I will not use docker cause sometime will not be well connected with ollama but go more for direct in terminal : 

```
sudo apt update
sudo apt full-upgrade -y
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt update
sudo apt install python3.11 -y
*** prepare python environment ***
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt update
sudo apt install python3.11 -y
source open-webui-env/bin/activate
pip install open-webui

****Lauch open webui **** 
open-webui serve 

****Ollama (Make sure ollama is install on ubuntu and running **** 
Ollama list 

*** Pull a model if necessary ( If you have no model listed after push command ollama list***
ollama pull llama3.2


go from server localhost:8080 for see the UI , unless you didn't install a desktop for UBUNTU server , then just run curl -I http://localhost:8080 to verify is active 

```

Then from you Mac : use  http://ipofyourserver:8080 to see the GUI , and normally on model list you shall yours

**Reply by TheMohawkNinja** [reply]
There needs to be a better resolution for docker users, especially given it is officially supported in the Open-WebUI GitHub repository. I am using Nvidia M10 GRID cards and have found that the self-contained docker container with build-in Nvidia support, Ollama, and Open-WebUI had been working perfectly fine out of the box until this issue occurred.

We need to be allowed to pull models from the web interface. @peuportier answer of installing an entire programming language and some unknown "deadsnakes" package is unacceptable.

**Reply by peuportier** [reply]
@TheMohawkNinja If you’re not comfortable using the DeadSnakes repository, you can download the Python binaries directly from the official Python website: https://www.python.org/downloads/. This is a perfectly acceptable alternative. Or do it per Docker

That said, the community generally trusts the DeadSnakes repository, so any concerns would be entirely your preference. You can install OWUI and Ollama in the way that works best for you. Detailed instructions for this method are also available in the OWUI documentation. Up to you.

**Comment by silvernode** [comment]
Discussion https://github.com/open-webui/open-webui/discussions/7054

Go to 

```
Admin Panel
    ➡️ Settings
        ➡️Connections
             ➡️ Manage Ollama API connections
                 ➡️ Manage (🔧 icon) 
```
![image](https://github.com/user-attachments/assets/1aed8eb8-bc90-4b35-af70-fc8e22711152)

**Reply by tjbck** [reply]
I'd highly encourage you to read the changelogs before updating.

**Comment by Smuzzies** [comment]
Any update on base model missing after upgrade? Or is there method of downgrading to a version that still works?

**Comment by Smuzzies** [comment]
![image](https://github.com/user-attachments/assets/fe1d3d7a-b47c-4fea-85e2-e013a271f272)

**Reply by peuportier** [reply]
By this way you can combine models?

**Comment by silvernode** [comment]
...because then I wouldn't think a feature was removed and I would know
that it was moved to a different menu.

**Comment by tounsicorp** [comment]
So what I see everyone saying.. Is that this system is useless because you either sign up with their account to use models (through boring json).. ORR  Attach a API that needs to run through other resources and not run locally..  Got it.  :D

**Comment by krausernexa** [comment]
so, doesn't mother which model you download from the website (openwebui.com), you always need to specify a base model which is llama3.2:70b ? i mean, if you download chatgpt4o it does run under model ollama3.2:70b? i think that's confusing a lot of people, if that's how this is supposed to be done ...

**Comment by viba1** [comment]
Same problem here: unable to download or upgrade any model on open-webui v0.5.4.
Neither using "admin parameters/parameters/models" nor using "workspace/models". 
Only import/export from a file and no possibility to upgrade a model.
After searching a long time, I finally found the desired menu:
- you have to go to "admin parameters/parameters/connexions" and then click on parameter gear icon.
-  for upgrading all models click on the "update all models" icon, hidden near the closing cross.



Since v0.4, the team added lot of useless configuration parameters which complicates significantly the interface and makes it boring like a huge monster gasworks.
In #7054 they try to justify: "Separation of Admin Rights and Responsibilities", but you already have an administrator account and the administration settings, but even more they decide to hide.

Please go back and simplify, or else we will fork…
Less is more!

**Reply by peuportier** [reply]
Yes I think they shall dissociate SUPER ADMIN who can give what right ADMIN can have and group, users. That would be more clear. Then remove useless options
